plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,408 @@
+            mac_sec_key=dict(
+                type="list",
+                elements="dict",
+                options=dict(
+                    key_name=dict(type="str", required=True),
+                    psk=dict(type="str", required=True, no_log=True),
+                    start_time=dict(type="str"),
+                    end_time=dict(type="str"),
+                ),
+                no_log=False,
+            ),
+            state=dict(type="str", choices=["absent", "query", "present"], default="query"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["mac_sec_policy"]],
+            ["state", "absent", ["mac_sec_policy"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    mac_sec_policy = module.params.get("mac_sec_policy")
+    mac_sec_policy_uuid = module.params.get("mac_sec_policy_uuid")
+    description = module.params.get("description")
+    admin_state = module.params.get("admin_state")
+    type = module.params.get("type")
+    cipher_suite = module.params.get("cipher_suite")
+    window_size = module.params.get("window_size")
+    security_policy = module.params.get("security_policy")
+    sak_expiry_time = module.params.get("sak_expiry_time")
+    confidentiality_offset = module.params.get("confidentiality_offset")
+    key_server_priority = module.params.get("key_server_priority")
+    mac_sec_keys = module.params.get("mac_sec_key")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_policy", template)
+    mso_template.validate_template("fabricPolicy")
+
+    path = "/fabricPolicyTemplate/template/macsecPolicies"
+
+    existing_mac_sec_policies = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {}).get("macsecPolicies", [])
+    if mac_sec_policy or mac_sec_policy_uuid:
+        object_description = "MACSec Policy"
+        if mac_sec_policy_uuid:
+            match = mso_template.get_object_by_uuid(object_description, existing_mac_sec_policies, mac_sec_policy_uuid)
+        else:
+            kv_list = [KVPair("name", mac_sec_policy)]
+            match = mso_template.get_object_by_key_value_pairs(object_description, existing_mac_sec_policies, kv_list)
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_mac_sec_policies
+
+    if state == "present":
+
+        mso.existing = {}
+
+        if match:
+
+            if mac_sec_policy and match.details.get("name") != mac_sec_policy:
+                ops.append(dict(op="replace", path="{0}/{1}/name".format(path, match.index), value=mac_sec_policy))
+                match.details["name"] = mac_sec_policy
+
+            if description and match.details.get("description") != description:
```

Comment:
what happens with empty description? can we set to empty?

Reply:
yes it still works if it's set to empty string

Reply:
I do not think you are correct on this that it works when you want to set empty string on update this way because empty string is a false condition. See snippet below and try it out

```
>>> details = {"description": "foo"}
>>> description = "bar"
>>> if description and details.get("description") != description:
...     print(description)
... else:
...     print("no match")
... 
bar
>>> description = ""
>>> if description and details.get("description") != description:
...     print(description)
... else:
...     print("no match")
... 
no match
```

So I think you should check if description is not None. The same would apply for any string value which should be possible to be reset to empty string

Reply:
https://github.com/CiscoDevNet/ansible-mso/blob/master/plugins/modules/ndo_route_map_policy_multicast.py#L216 similar situation